app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,27 @@
+class RecipesController < ApplicationController
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.create(recipe_params)
+    if @recipe.save
+      flash[:notice] = "Recipe saved."
+      redirect_to @recipe
+    else
+      flash[:error] = "There was an error saving your recipe: #{@recipe.errors.full_messages.first}"
+      redirect_to new_recipe_path
```

Comment:
I tried this, but it's not showing the 3 grains fields when there's a validation error. For some reason, the `3.times { @recipe.grains.build }` that's in the controller's recipe#new method isn't firing on `render :new`